sigmastate/src/main/scala/sigmastate/lang/syntax/Exprs.scala

```diff
@@ -30,74 +30,76 @@ trait Exprs extends Core with Types {
   // Expressions directly within a `val x = ...` or `def x = ...`
   object FreeCtx extends WsCtx(semiInference=true, arrowTypeAscriptions=true)
 
-  val TypeExpr = ExprCtx.Expr
+  def TypeExpr[_:P] = ExprCtx.Expr
 
   private val predefFuncRegistry = new PredefinedFuncRegistry(builder)
   import predefFuncRegistry._
 
   //noinspection TypeAnnotation,ForwardReference
   class WsCtx(semiInference: Boolean, arrowTypeAscriptions: Boolean){
 
-    val OneSemiMax = if (semiInference) OneNLMax else Pass
-    val NoSemis = if (semiInference) NotNewline else Pass
+    def OneSemiMax[_:P] = if (semiInference) OneNLMax else Pass
+    def NoSemis[_:P] = if (semiInference) NotNewline else Pass
 
-    val Expr: P[Value[SType]] = {
-      val If = {
-        val Else = P( Semi.? ~ `else` ~/ Expr )
+    def Expr[_:P]: P[Value[SType]] = {
+      def If = {
+        def Else = P( Semi.? ~ `else` ~/ Expr )
         P( Index ~ `if` ~/ "(" ~ ExprCtx.Expr ~ ")" ~ Expr ~ Else ).map {
           case (i, c, t, e) => atSrcPos(i) { mkIf(c.asValue[SBoolean.type], t, e) }
         }
       }
-      val Fun = P(`def` ~ FunDef)
+      def Fun = P(`def` ~ FunDef)
 
-      val LambdaRhs = if (semiInference) P( (Index ~ BlockChunk).map {
+      def LambdaRhs = if (semiInference) P( (Index ~ BlockChunk).map {
```

Comment:
ScalaDoc missed here and everywhere in this trait

Reply:
fixed in https://github.com/ScorexFoundation/sigmastate-interpreter/commit/2cee01a333ebfa83b24d849b223d5eb89db8bb52